packages/features/data-table/DataTableProvider.tsx

```diff
@@ -112,16 +119,23 @@ export function DataTableProvider({
 
   const removeFilter = useCallback(
     (columnId: string) => {
-      setPageIndex(0);
-      setActiveFilters((prev) => prev.filter((filter) => filter.f !== columnId));
+      // To avoid showing empty/default data in the query string using `null` ,
+      // Nuqs says: Setting null as a value will remove the key from the query string.
+      setPageIndex(null);
+      setActiveFilters((prev) => {
+        const remainingFilters = prev.filter((filter) => filter.f !== columnId);
+        return remainingFilters.length == 0 ? null : remainingFilters;
+      });
     },
     [setActiveFilters]
   );
 
   const setPageSizeAndGoToFirstPage = useCallback(
-    (newPageSize: number) => {
-      setPageSize(newPageSize);
-      setPageIndex(0);
+    (newPageSize: number | null) => {
+      // To avoid showing empty/default data in the query string using `null` ,
+      // Nuqs says: Setting null as a value will remove the key from the query string.
+      setPageSize(newPageSize == DEFAULT_PAGE_SIZE ? null : newPageSize);
+      setPageIndex(null);
     },
     [setPageSize, setPageIndex]
   );
```

Comment:
Can we do something like this for setPageIndex also?

Instead of providing a raw `setPageIndex` to the context, it'd be nice to provide a wrapper function that takes care of `null`.